rickchurch/church.py

```diff
@@ -105,22 +110,37 @@ async def authorize() -> fastapi.Response:
 @app.get("/oauth_callback", include_in_schema=False)
 async def auth_callback(request: fastapi.Request) -> fastapi.Response:
     """This endpoint is only used as a redirect target from discord OAuth2."""
+    client = request.app.state.client
     code = request.query_params["code"]
     try:
-        user = await get_oauth_user(code)
+        user, access_token = await get_oauth_user(client, code)
         token = await add_user(user, request.state.db_conn)
     except PermissionError:
         # `add_user` can return `PermissionError` if the user already has a token, which is banned.
         raise fastapi.HTTPException(401, "You are banned")
 
+    # Join them into the rickchurch server
+    res = await client.put(
+        constants.DISCORD_BASE_URL + f"/guilds/{constants.discord_guild_id}" + f"/members/{user['id']}",
+        json={"access_token": access_token},
+        headers={"Authorization": f"Bot {constants.discord_bot_token}"},
+    )
+    if res.status_code not in [200, 201, 204]:
```

Comment:
Shouldn't we only handle for 1 of these codes? How come we can get any of these 3?

If for some reason we really should be handling for all 3, tuple would make more sense here than a list. There should also be a comment explaining why it is necessary to handle all 3 of these codes, since it isn't very obvious.

Reply:
Using a tuple sounds good. I will add a comment but 201 means they joined and 204 means they were already in the server.